src/middleware/authMiddleware.js

```
@@ -0,0 +1,29 @@
+const jwt = require("jsonwebtoken");
+const { UnauthenticatedError } = require("../errors");
+
+const authMiddleware = (req, res, next) => {
+  //check header
+  const authHeader = req.headers.authorization;
+  if (
+    !authHeader ||
+    !authHeader.startsWith("Bearer ") ||
+    authHeader.split(" ")[1] === "null"
+  ) {
+    throw new UnauthenticatedError("Authentication invalid");
+  }
+  const token = authHeader.split(" ")[1];
+  try {
+    const payload = jwt.verify(token, process.env.JWT_SECRET);
+    //attach the user to the data routes
+    req.user = { userId: payload.userId, name: payload.name };
+    next();
+  } catch (error) {
+    if (err.name === "TokenExpiredError") {
+      throw new UnauthenticatedError("Authentication invalid");
+    } else {
+      throw new UnauthenticatedError("Authentication invalid");
```

You don't need this if you are going to throw the same error. If you want to throw a different error when the token is expired then you can keep the if and change the error message.